### ISSUE TITLE
gnt-job info and list corrupted after gnt-cluster destroy / init

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#970]
Originally reported of Google Code with ID 914.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

# gnt-cluster --version
gnt-cluster (ganeti v2.10.1) 2.10.1

# gnt-cluster version
Software version: 2.10.1
Internode protocol: 2100000
Configuration format: 2100000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.10.1

# hspace --version
hspace (ganeti) version v2.10.1
compiled with ghc 6.12
running on linux x86_64

<b>What distribution are you using?</b>
debian

<b>What steps will reproduce the problem?</b>
1. gnt-cluster init ...
2. gnt-node add ...
3. run jobs.
4. gnt-node remove ..
5. gnt-cluster destroy ...
6. gnt-cluster init ..., gnt-node add ..., run jobs..
7. gnt-job info or gnt-job list


<b>What is the expected output? What do you see instead?</b>

The expected output is the most recent job information. However, we observed that the job list and information was being reported from /var/lib/ganeti/queue/archive/0/ (jobs from previous cluster state before the latest gnt-cluster init) instead of /var/lib/ganeti/queue/ (jobs since the latest gnt-cluster init). This was confusing and caused automation failures because the expected jobs appeared to not exist after submission.

We observe that /var/lib/ganeti/queue/serial was reset across cluster inits, but /var/lib/ganeti/queue/archive/ was not cleaned up. If there was a monotonically increasing serial number or archive directory was cleaned up, then the job list will not cause confusion.

This bug is a request for gnt-cluster init / destroy / init to leave the system in a consistent state.

```
Originally added on 2014-08-08 20:27:56 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-08-12 07:26:56 +0000 UTC.

Added Labels: Type-Defect Priority-High
Changed State: Accepted
Added to Milestone: Release2.13

**Comment by Ganeti-Issues-Migrator** [comment]
```
After some offline discussion we found that simply
removing those files is dangerous (as it might lead
to loss of data relevant for auditing what happened).
So our plan is to add a check to cluster init that
refuses to init the cluster if old data is still present
telling the user to remove the files if he is sure that
they are no longer needed.

Would that solution work for you as well?
```
Originally added on 2014-08-18 13:18:06 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 3a8056893b432e918a41a28ed4f3e1c81a0734fb
Author: Klaus Aehlig <aehlig@google.com>
Date:   Mon Aug 18 17:57:32 2014 +0200

    Verify clean DATA_DIR on cluster init
    
    A new Ganeti cluster should not inherit state from a previously existing
    cluster. Therefore, on cluster initialization, verify that no dangerous
    left-overs are present in the DATA_DIR. We cannot, however, insist on
    the DATA_DIR being completely empty, as it is good practice to set up
    the rapi user file ahead of time. What we therefore do is to verify
    that
    - the DATA_DIR, if present, only contains directory entries, and
    - the QUEUE_DIR, if present, is empty.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2014-08-18 16:37:24 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking as fixed now. Feel free to reopen, if just checking
(and hence insisting on the user cleaning up himself) is not
enough.
```
Originally added on 2014-08-18 16:38:05 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Thank you!
```
Originally added on 2014-08-18 16:46:36 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:37:02 +0000 UTC.

Changed State: Released
